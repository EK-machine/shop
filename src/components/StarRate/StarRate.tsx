import React from 'react';
import styles from './style.module.css';
import { StarRateProps } from '../../interface/intefaces';

const StarRate: React.FC<StarRateProps> = ({ rating }) => (
  <div className={styles.starContainer}>
    {[1, 2, 3, 4, 5].map((star, i) => (
      <svg key={star} width="16" height="15" viewBox="0 0 16 15" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path
          d="M8.31034 0.375L6.00805 5.13533L0.862061 5.89429L4.5862 9.60136L3.70576 14.8315L8.31034 12.3636L12.9149 14.8315L12.0345 9.60136L15.7586 5.89946L10.6126 5.13533L8.31034 0.375Z"
          fill={i < rating ? '#f5d442' : '#F5F5F5'}
        />
      </svg>
    ))}
  </div>
);

export default StarRate;
