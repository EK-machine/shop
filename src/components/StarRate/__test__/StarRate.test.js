import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import StarRate from '../StarRate';

describe('Component: StarRate', () => {
  test('renders the stars', () => {
    render(<StarRate rating={2} />);
    const gold = screen.getAllByTestId('gold');
    const grey = screen.getAllByTestId('grey');
    const stars = [...gold, ...grey];
    expect(stars).toBeTruthy();
  });

  test('renders 5 stars', () => {
    render(<StarRate rating={3} />);
    const gold = screen.getAllByTestId('gold');
    const grey = screen.getAllByTestId('grey');
    const stars = [...gold, ...grey];
    expect(stars.length).toEqual(5);
  });

  test('renders 1 grey and 4 golden stars', () => {
    const num = 4;
    render(<StarRate rating={num} />);
    const gold = screen.getAllByTestId('gold');
    const grey = screen.getAllByTestId('grey');
    expect(gold.length).toEqual(num);
    expect(grey.length).toEqual(1);
  });

  test('it renders correctly (snapshot)', () => {
    const tree = renderer.create(<StarRate rating={5} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
