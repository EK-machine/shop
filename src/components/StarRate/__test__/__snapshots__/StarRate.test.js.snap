// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: StarRate it renders correctly (snapshot) 1`] = `
<div
  className="starContainer"
>
  <svg
    data-testid="gold"
    fill="none"
    height="15"
    viewBox="0 0 16 15"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.31034 0.375L6.00805 5.13533L0.862061 5.89429L4.5862 9.60136L3.70576 14.8315L8.31034 12.3636L12.9149 14.8315L12.0345 9.60136L15.7586 5.89946L10.6126 5.13533L8.31034 0.375Z"
      fill="#f5d442"
    />
  </svg>
  <svg
    data-testid="gold"
    fill="none"
    height="15"
    viewBox="0 0 16 15"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.31034 0.375L6.00805 5.13533L0.862061 5.89429L4.5862 9.60136L3.70576 14.8315L8.31034 12.3636L12.9149 14.8315L12.0345 9.60136L15.7586 5.89946L10.6126 5.13533L8.31034 0.375Z"
      fill="#f5d442"
    />
  </svg>
  <svg
    data-testid="gold"
    fill="none"
    height="15"
    viewBox="0 0 16 15"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.31034 0.375L6.00805 5.13533L0.862061 5.89429L4.5862 9.60136L3.70576 14.8315L8.31034 12.3636L12.9149 14.8315L12.0345 9.60136L15.7586 5.89946L10.6126 5.13533L8.31034 0.375Z"
      fill="#f5d442"
    />
  </svg>
  <svg
    data-testid="gold"
    fill="none"
    height="15"
    viewBox="0 0 16 15"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.31034 0.375L6.00805 5.13533L0.862061 5.89429L4.5862 9.60136L3.70576 14.8315L8.31034 12.3636L12.9149 14.8315L12.0345 9.60136L15.7586 5.89946L10.6126 5.13533L8.31034 0.375Z"
      fill="#f5d442"
    />
  </svg>
  <svg
    data-testid="gold"
    fill="none"
    height="15"
    viewBox="0 0 16 15"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.31034 0.375L6.00805 5.13533L0.862061 5.89429L4.5862 9.60136L3.70576 14.8315L8.31034 12.3636L12.9149 14.8315L12.0345 9.60136L15.7586 5.89946L10.6126 5.13533L8.31034 0.375Z"
      fill="#f5d442"
    />
  </svg>
</div>
`;
